sound/soc/sof/intel/hda-dai.c

```diff
@@ -482,6 +487,26 @@ int sdw_hda_dai_hw_params(struct snd_pcm_substream *substream,
 	int ret;
 	int i;
 
+	ops = hda_dai_get_ops(substream, cpu_dai);
+	if (!ops) {
+		dev_err(cpu_dai->dev, "DAI widget ops not set\n");
+		return -EINVAL;
+	}
+
+	sdev = widget_to_sdev(w);
+	hext_stream = ops->get_hext_stream(sdev, cpu_dai, substream);
+	if (hext_stream && hext_stream->link_prepared)
+		return 0;
+
+	/* reset the PCMSyCM registers */
+	ret = hdac_bus_eml_sdw_map_stream_ch(sof_to_bus(sdev), link_id, cpu_dai->id,
```

Comment:
This warrants a bit more explanation. Why is this added? This looks a bit like a seprate bugfix if this is an action that needs to be done in prepare, or this is this directly related to reseting the link DMA channe? In either way, a comment would help.

Reply:
Yeah, we could comment this inline a bit more to allow the decision making flow to be more readable.